cabinets/cabinet/s3_cabinet.py

```diff
@@ -64,3 +66,28 @@ def delete_content(cls, path, **kwargs):
         except Exception as ex:
             error(f"Cannot delete {path} from S3 Bucket '{bucket}': {ex}")
             return False
+
+    @classmethod
+    def ls(cls, directory: str, delimiter: str = '/', **kwargs) -> List[str]:
+        cls._ensure_client_exists()
+
+        bucket, *dir = directory.split(delimiter)
+        dir = '/'.join(dir)
+
+        files = []
+        response = cls.client.list_objects(Bucket=bucket, Prefix=dir)
```

Comment:
We probably want to be using [list_objects_v2](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.list_objects_v2) right?

Reply:
Yeah I was unsure about this. Is v2 just an updated version of the same thing?

Reply:
[This](https://stackoverflow.com/a/37539994/6121611) captures the meat of it.